addons/mail/tests/discuss/test_discuss_channel_member.py

```diff
@@ -28,133 +28,32 @@ def setUpClass(cls):
-        channel_members = self.env['discuss.channel.member'].search([('channel_id', '=', self.group.id)])
-        self.assertEqual(len(channel_members), 1)
-
-        # User 2 is not in the group, they can not invite user 3
-        with self.assertRaises(AccessError):
-            self.env['discuss.channel.member'].with_user(self.user_2).create({
-                'partner_id': self.user_portal.partner_id.id,
-                'channel_id': self.group.id,
-            })
-
-        # User 1 is in the group, they can invite other users
-        self.env['discuss.channel.member'].with_user(self.user_1).create({
-            'partner_id': self.user_portal.partner_id.id,
-            'channel_id': self.group.id,
-        })
-        channel_members = self.env['discuss.channel.member'].search([('channel_id', '=', self.group.id)])
-        self.assertEqual(channel_members.mapped('partner_id'), self.user_1.partner_id | self.user_portal.partner_id)
-
-        # But User 3 can not write on the `discuss.channel.member` of other user
-        channel_member_1 = self.env['discuss.channel.member'].search([('channel_id', '=', self.group.id), ('partner_id', '=', self.user_1.partner_id.id)])
-        channel_member_3 = self.env['discuss.channel.member'].search([('channel_id', '=', self.group.id), ('partner_id', '=', self.user_portal.partner_id.id)])
-        channel_member_3.with_user(self.user_portal).custom_channel_name = 'Test'
+            member.channel_id = another_channel
         with self.assertRaises(AccessError):
-            channel_member_1.with_user(self.user_2).custom_channel_name = 'Blabla'
-        self.assertNotEqual(channel_member_1.custom_channel_name, 'Blabla')
-
-    def test_group_invite(self):
-        """Test invitation in group part 2 (use `invite` action)."""
-        self.group.with_user(self.user_1).sudo()._add_members(users=self.user_1)
-        channel_members = self.env['discuss.channel.member'].search([('channel_id', '=', self.group.id)])
-        self.assertEqual(channel_members.mapped('partner_id'), self.user_1.partner_id)
-
-        # User 2 is not in the group, they can not invite user_portal
+            member.partner_id = another_partner
         with self.assertRaises(AccessError):
-            self.group.with_user(self.user_2)._add_members(users=self.user_portal)
-        channel_members = self.env['discuss.channel.member'].search([('channel_id', '=', self.group.id)])
-        self.assertEqual(channel_members.mapped('partner_id'), self.user_1.partner_id)
+            member.guest_id = guest
 
-        # User 1 is in the group, they can invite user_portal
-        self.group.with_user(self.user_1)._add_members(users=self.user_portal)
-        channel_members = self.env['discuss.channel.member'].search([('channel_id', '=', self.group.id)])
-        self.assertEqual(channel_members.mapped('partner_id'), self.user_1.partner_id | self.user_portal.partner_id)
-
-    def test_group_leave(self):
```

Comment:
This test ensures members cannot unlink others. Better coverage in `test_10_discuss_channel_member_access`: ` ("user", "group", "member", "other", "unlink", False)`.